Source/Scene/PostProcessDepthOfFieldStage.js

```diff
@@ -0,0 +1,125 @@
+define([
+        '../Core/Check',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Shaders/PostProcessFilters/DepthOfField',
+        '../Shaders/PostProcessFilters/GaussianBlur1D',
+        './PostProcess',
+        './PostProcessBlurStage',
+        './PostProcessComposite'
+    ], function(
+        Check,
+        defineProperties,
+        destroyObject,
+        DepthOfField,
+        GaussianBlur1D,
+        PostProcess,
+        PostProcessBlurStage,
+        PostProcessComposite) {
+    'use strict';
+
+    /**
+     * Post process stage for depth of field. Implements {@link PostProcess}.
+     *
+     * @alias PostProcessDepthOfFieldStage
+     * @constructor
+     *
+     * @private
+     */
+    function PostProcessDepthOfFieldStage() {
+        this._name = 'czm_depth_of_field';
+
+        this._blurProcess = new PostProcessBlurStage({
+            name : 'czm_depth_of_field_blur'
+        });
+        var depthOfFieldProcess = new PostProcess({
+            name : 'czm_depth_of_field_composite',
+            fragmentShader : DepthOfField,
+            uniformValues : {
+                focalDistance : 5.0,
+                blurTexture : this._blurProcess.name
+            }
+        });
+        this._composite = new PostProcessComposite({
+            name : 'czm_depth_of_field_composite',
+            processes : [this._blurProcess, depthOfFieldProcess],
```

Comment:
I don't know about our exact implementation, but can't this be done in one pass where the blur kernel depends on the pixels depth?  Looks like this approach blurs everything and then composites it based on the distance.

Maybe a roadmap item if this DOF looks reasonable enough and is fast enough.